src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -178,16 +183,28 @@ async def _handle_unary_unary_rpc(object method_handler,
     )
 
     # Sends response message
-    cdef tuple send_ops = (
-        SendStatusFromServerOperation(
-            tuple(),
-            StatusCode.ok,
-            b'',
-            _EMPTY_FLAGS,
-        ),
-        SendInitialMetadataOperation(None, _EMPTY_FLAGS),
-        SendMessageOperation(response_raw, _EMPTY_FLAGS),
-    )
+    cdef tuple send_ops
+    if rpc_state.metadata_sent:
```

Comment:
There's a lot of similarity between this sequence of operations and the one in the next block. You can construct the second by just prepending a `SendInitialMetadata` operation to the first.